test/unit/hello.world.controller.test.ts

```diff
@@ -0,0 +1,72 @@
+import { HelloWorldController } from "../../src/controllers/hello-world";
+import { UserProfile } from "@loopback/authentication";
+import { expect, sinon } from "@loopback/testlab";
+import {
+  DefaultCrudRepository,
+  DataSourceConstructor
+} from "@loopback/repository";
+import { UserRepository } from "../../src/repositories/user.repository";
+
+describe("HelloWorldController", () => {
+  let repository: UserRepository;
+  let controller: HelloWorldController;
+  let testUserProfile: UserProfile = {
+    id: "1",
+    name: "foo",
+    email: "foo@bar.com"
+  };
+
+  beforeEach(givenController);
+
+  context("helloWorld()", () => {
+    it("returns Hello World message for a given user", async () => {
+      let stub = repository.create as sinon.SinonStub;
+      const res = await controller.helloWorld("Adam");
+      expect.ok(repository.create, "create was called");
+      expect.deepEqual(
+        stub.getCall(0).args,
+        [{ username: "Adam" }],
+        "args were correct"
+      );
+      expect(res).to.equal(
+        "Hello world Adam " + JSON.stringify(testUserProfile)
+      );
+    });
+
+    it("returns undefined when user not specified", async () => {
+      let stub = repository.create as sinon.SinonStub;
+      const res = await controller.helloWorld();
+      expect.ok(stub.called, "create was called");
+      expect.deepEqual(
```

Comment:
ditto, use `expect(stub).to.be.calledWithMatch` instead of both assertions (L39 and this one)